lib/engine.py

```diff
@@ -706,7 +706,8 @@ def filename_from_url(url):
 
 
 def gen_from_source(source):
-    """Returns a generator from a source tuple.
+    """
+    Returns a generator from a source tuple.
```

Comment:
PEP257 says it should be as it was originally: https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings